Code/PgSQL/rdkit/rdkit_gist.c

```diff
@@ -535,6 +536,41 @@ Datum gmol_consistent(PG_FUNCTION_ARGS) {
   PG_RETURN_BOOL(res);
 }
 
+
+/*
+ * Sortsupport function
+ *
+ * Returns a comparator function to sort data in a way that preserves locality.
+ */
+static int
+gmol_cmp(Datum x, Datum y, SortSupport ssup)
+{
+  /* establish order between x and y */
+  bytea *a = (bytea*)DatumGetPointer(PG_DETOAST_DATUM(x));
+  bytea *b = (bytea*)DatumGetPointer(PG_DETOAST_DATUM(y));
+
+  Assert(!ISALLTRUE(a));
+  Assert(!ISALLTRUE(b));
+
+  int siglen = SIGLEN(a);
+  Assert(siglen == SIGLEN(b));
+
+  int retval = bitstringGrayCmp(siglen, VARDATA(a), VARDATA(b));
+  RDKIT_FREE_IF_COPY_P(a, x);
+  RDKIT_FREE_IF_COPY_P(b, y);
```

Comment:
the extra copies created by detoasting are usually fine to ignore and assume to be automatically reclaimed, but this comparison function may be called a huge number of times when an index is built, making the explicit deallocation necessary.